kafka-client/src/main/java/dev/responsive/utils/CachedMetadata.java

```diff
@@ -0,0 +1,68 @@
+package dev.responsive.utils;
+
+import static dev.responsive.db.ColumnName.PARTITION_KEY;
+
+import com.datastax.oss.driver.api.core.cql.ResultSet;
+import dev.responsive.db.CassandraClient;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.OptionalInt;
+import java.util.concurrent.ExecutionException;
+import org.apache.kafka.clients.admin.Admin;
+
+/**
+ * Use this any time we need to repeatedly look up metadata that doesn't change, so we can
+ * cache the results rather than make admin/client requests every single time
+ */
+public class CachedMetadata {
```

Comment:
This class seems to be shared by all the stores, so it needs to be synchronized.